frame/democracy/src/lib.rs

```diff
@@ -596,14 +612,28 @@ decl_module! {
 		/// - Db reads: `PublicPropCount`, `PublicProps`
 		/// - Db writes: `PublicPropCount`, `PublicProps`, `DepositOf`
 		/// # </weight>
-		#[weight = T::WeightInfo::propose()]
-		fn propose(origin, proposal_hash: T::Hash, #[compact] value: BalanceOf<T>) {
+		#[weight = T::WeightInfo::propose(*prop_count)]
+		fn propose(origin,
+			proposal_hash: T::Hash,
+			#[compact] value: BalanceOf<T>,
+			#[compact] prop_count: u32,
```

Comment:
why not documenting the new param `prop_count` in the doc ?

db reads in the doc is wrong, it needs one more read of Blacklist, should be better to remove it or update it.

Also it is weird to see `O(1)` while benchmark result in `O(p)`. I see that this `O(p)` is not a cost inside runtime but only inside database (in operation append and maybe also decode_len).